shm/Map.cpp

```diff
@@ -0,0 +1,51 @@
+
+#include <iostream>
+
+#include "Map.hpp"
+#include "ShmExceptions.hpp"
+
+Map::Map() : mapXsize(defaultXsize), mapYsize(defaultYsize)
+{
+    randomIslandGenerator(defaultIslandsAmount);
+}
+Map::Map(int sizeY, int sizeX, int islandsAmount) : mapXsize(sizeY), mapYsize(sizeX)
```

Comment:
mapXsize and mapYsize are size_t, while sizeX and sizeY are int type

btw. you initialize mapXsize with sizeY and mapYsize with sizeX values - x with y and y with x.

Reply:
+1